src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/Grid.cs

```diff
@@ -297,6 +308,15 @@ public RowDefinitionCollection RowDefinitions
 
                 return (_data.RowDefinitions);
             }
+            set
+            {
+                if (value == null){
+                    _data.RowDefinitions = new RowDefinitionCollection(this);
+                    return;
+                }
+                _data ??= new ExtendedData();
```

Comment:
I didn't check how _data gets initialized, but if this is supposed to be lazy-init, you get NRE if `value` was `null`.

Reply:
Fixed it